src/spark-core.js

```diff
@@ -144,9 +144,9 @@ var SparkCore = AmpersandState.extend({
     return this.device.remove()
       .catch(function logDeviceRemovalFailure(reason) {
         this.logger.warn(reason);
-      })
+      }.bind(this))
       .then(function clearEncryptStore() {
-        this.encryption.keystore.clear();
+        return this.encryption.keystore.clear();
```

Comment:
@ianwremmel, is encryption always defined in legacy sdk or should we also check `if (this.encryption)`